log4j-core/src/main/java/org/apache/logging/log4j/core/util/Constants.java

```diff
@@ -76,8 +76,10 @@ public final class Constants {
      * {@code true} if we think we are running in a web container, based on the boolean value of system property
      * "log4j2.is.webapp", or (if this system property is not set) whether the  {@code javax.servlet.Servlet} class
      * is present in the classpath.
+     * @deprecated use {@link org.apache.logging.log4j.util.Constants#isWebApp()}
      */
-    public static final boolean IS_WEB_APP = org.apache.logging.log4j.util.Constants.IS_WEB_APP;
+    @Deprecated
```

Comment:
Why are we adding deprecated fields?

Reply:
I think we are allowed to break binary backward compatibility in non-API[1] classes. Otherwise, doesn't this conflict with the rationale of a major release? Maybe this is indeed a gray area and we should discuss further. Though my stance is, as long as it is improving the architecture and not breaking the API[1], go ahead.

[1] For me, API translates to the `log4j-api` and the file-based configuration API.